src/WordPress/Blueprints/Resources/Resolver/InlineResourceResolver.php

```diff
@@ -9,10 +9,10 @@
 
 class InlineResourceResolver implements ResourceResolverInterface {
 
-	public function parseUrl( string $url ): ResourceDefinitionInterface|false {
+	public function parseUrl( string $url ): ?ResourceDefinitionInterface {
 		// If url starts with "protocol://" then we assume it's not inline raw data
 		if ( 0 !== preg_match( '#^[a-z_+]+://#', $url ) ) {
-			return false;
+			return null;
```

Comment:
Why change this logic?

Reply:
PHP version support.

`?ResourceDefinitionInterface` is fine for PHP 7.0 (7.4)
`ResourceDefinitionInterface|false` is not - since union type are not allowed

`null` is still falsy. I would love to return an optional though...

Reply:
This change seems sound so let's roll with it, but in general it's fine to omit the type annotation entirely if the language cannot express something.

Reply:
Optionals are not that popular in PHP and this library will be used by developers with different skillsets. Let's avoid introducing too many ideas.

Reply:
Ok on the optionals front. Yet, I think cohesive typing is a valid goal, especially for the reason that makes you reluctant to introduce new concepts - accessibility for dev with different skillsets.